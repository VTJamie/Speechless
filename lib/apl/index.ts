export * from './commands';
export * from './components';
export * from './document';
export * from './filters';
export * from './interfaces';
export * from './avg';
export * from './gestures';
