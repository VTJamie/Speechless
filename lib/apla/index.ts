export * from './document';
export * from './interfaces';
export * from './components';
