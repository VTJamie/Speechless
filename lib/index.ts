export * as skill from './skill';
export * as apl from './apl';
export * as apla from './apla';
// export * as acdl from './acdl';
export * from './types';
