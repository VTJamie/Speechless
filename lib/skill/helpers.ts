import * as Alexa from 'ask-sdk-core';
import { Request } from 'ask-sdk-model';
import { SHA256 } from 'crypto-js';

import { IntentStatusProps } from './models';
import { IntentHandler } from './types';

export const wrapHandler = <T extends Request>(handler: IntentHandler<T>) => {
  return (input: Alexa.HandlerInput) => {
    const contextApl = input.requestEnvelope.context['Alexa.Presentation.APL'];
    const status: IntentStatusProps = {
      displaySupported: Alexa.getSupportedInterfaces(input.requestEnvelope)['Alexa.Presentation.APL'] ? true : false,
      locale: Alexa.getLocale(input.requestEnvelope),
      aplDocumentToken: contextApl?.token,
      attributesManager: input.attributesManager
    };
    const request = Alexa.getRequest(input.requestEnvelope);
    return handler(input, status, request as T);
  };
};

export const hashValue = (prompt: unknown): string => {
  return SHA256([JSON.stringify(prompt)].join('-'))
    .toString()
    .substring(0, 100);
};
