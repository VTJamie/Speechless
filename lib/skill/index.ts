export * from './intent';
export * from './locale';
export * from './models';
export * from './skills';
export * from './types';
