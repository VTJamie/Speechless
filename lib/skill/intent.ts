import * as Alexa from 'ask-sdk';
import { IntentRequest } from 'ask-sdk-model';

import { hashValue, wrapHandler } from './helpers';
import { SlotProps, ConfirmationProps, RequestHandlerProvider, SlotModel, LabeledRequestHandler, IntentPrompt } from './models';
import { IntentHandler, DelegationStrategyType } from './types';
import { ModelProvider } from '../types';

export interface IntentModel {
  /**
   * Name to identify the intent.
   */
  name: string;
  /**
   * Name of the generator used to generate this object.
   */
  generatedBy?: string;
  /**
   * List of slots within the intent.
   */
  slots?: Array<SlotModel>;
  /**
   * Phrases the user can speak e.g. to trigger an intent or provide value for a slot elicitation.
   */
  samples?: Array<string>;

  confirmationRequired?: boolean;
  prompts?: IntentPrompt;
  delegationStrategy?: DelegationStrategyType;
}

export interface IntentProps {
  intentName: string;
  intentHandler: IntentHandler<IntentRequest>;
  samples: string[];
  launchHandler?: boolean;
  slots?: SlotProps[];
  confirmation?: ConfirmationProps;
  delegationStrategy?: DelegationStrategyType;
}
export class Intent extends ModelProvider<IntentModel> implements RequestHandlerProvider {
  public readonly props: IntentProps;

  constructor(props: IntentProps) {
    super();

    this.props = props;
  }

  toJSON(): IntentModel {
    return this.model();
  }

  private getSlotModel(): SlotModel[] {
    return this.props.slots
      ? this.props.slots?.map((slot) => {
          return {
            name: slot.name,
            multipleValues: slot.multipleValues,
            samples: slot.samples,
            type: slot.type,
            confirmationRequired: slot.confirmation !== undefined,
            elicitationRequired: slot.required !== undefined,
            prompts: {
              confirmation: slot.confirmation ? hashValue(slot.confirmation.prompts) : undefined,
              elicitation: slot.required ? hashValue(slot.required.prompts) : undefined
            }
          };
        })
      : [];
  }

  model(): IntentModel {
    const intentModel: IntentModel = {
      name: this.props.intentName,
      slots: this.getSlotModel(),
      samples: this.props.samples,
      delegationStrategy: this.props.delegationStrategy
    };
    return intentModel;
  }

  usesDialog(): boolean {
    const countsOfDialogSlots = this.props.slots
      ? this.props.slots?.filter((slot) => {
          return slot.confirmation !== undefined || slot.required !== undefined;
        })
      : [];
    return this.props.confirmation !== undefined && countsOfDialogSlots.length > 0;
  }

  slotsUsingDialog(): SlotProps[] {
    return this.props.slots
      ? this.props.slots?.filter((slot) => {
          return slot.required !== undefined || slot.confirmation !== undefined;
        })
      : [];
  }

  getRequestHandlers(): LabeledRequestHandler[] {
    const handler: LabeledRequestHandler = {
      uniqueId: this.props.intentName,
      canHandle: (input: Alexa.HandlerInput) => {
        const requestType = Alexa.getRequestType(input.requestEnvelope);
        const canHandle = requestType === 'IntentRequest' && Alexa.getIntentName(input.requestEnvelope) === this.props.intentName;
        return canHandle;
      },
      handle: wrapHandler(this.props.intentHandler)
    };
    return [handler];
  }
}
