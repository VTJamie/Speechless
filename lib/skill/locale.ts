import { HandlerInput, getLocale, getRequestType, getIntentName } from 'ask-sdk-core';
import { IntentRequest, LaunchRequest, Request, SessionEndedRequest } from 'ask-sdk-model';
import * as fs from 'fs';

import { hashValue, wrapHandler } from './helpers';
import { Intent } from './intent';
import {
  InteractionModel,
  FilePackageWriter,
  RequestHandlerProvider,
  DialogIntents,
  DialogSlotItems,
  PromptItems,
  LabeledErrorHandler,
  LabeledRequestHandler
} from './models';
import { IntentHandler, JSBErrorHandler } from './types';
import { LocaleString, ModelProvider } from '../types';

export interface LocaleProps {
  locale: LocaleString;
  invocationName: string;
  cancelHandler?: IntentHandler<IntentRequest>;
  helpHandler?: IntentHandler<IntentRequest>;
  fallbackHandler?: IntentHandler<Request>;
  launchHandler?: IntentHandler<LaunchRequest>;
  sessionEndedHandler?: IntentHandler<SessionEndedRequest>;
  errorHandler?: JSBErrorHandler;
}
export class Locale extends ModelProvider<InteractionModel> implements FilePackageWriter, RequestHandlerProvider {
  readonly localeValue: LocaleString;
  readonly props: LocaleProps;

  readonly intents: { [intentName: string]: Intent };
  constructor(props: LocaleProps) {
    super();
    this.localeValue = props.locale;
    this.props = props;
    this.intents = {};
    this.props.cancelHandler = this.props.cancelHandler
      ? this.props.cancelHandler
      : (input: HandlerInput) => {
          console.log('Inside default cancel handler');
          return input.responseBuilder.speak('Goodbye').reprompt('Until again!').getResponse();
        };
    this.props.sessionEndedHandler = this.props.sessionEndedHandler
      ? this.props.sessionEndedHandler
      : (input: HandlerInput) => {
          console.log('Inside default session ended handler');
          return input.responseBuilder.speak('Ending session, goodbye').reprompt('Until again!').getResponse();
        };
    this.props.fallbackHandler = this.props.fallbackHandler
      ? this.props.fallbackHandler
      : (input: HandlerInput) => {
          console.log('Inside default fallbackHandler');
          return input.responseBuilder.speak("I am sorry I don't understand.").reprompt('Fell through!').getResponse();
        };
    this.props.helpHandler = this.props.helpHandler
      ? this.props.helpHandler
      : (input: HandlerInput) => {
          console.log('Inside default help handler');
          return input.responseBuilder.speak("I am sorry I can't help you.").reprompt("I am sorry I can't help you.").getResponse();
        };
    this.props.launchHandler = this.props.launchHandler
      ? this.props.launchHandler
      : (input: HandlerInput) => {
          console.log('Inside default Launch Handler');
          return input.responseBuilder.speak('Hello, welcome to the skill!').reprompt('Welcome').getResponse();
        };
    this.props.errorHandler = this.props.errorHandler
      ? this.props.errorHandler
      : (input: HandlerInput, error: Error) => {
          console.error('Inside default Error Handler', JSON.stringify(error), JSON.stringify(input), error);
          return input.responseBuilder.speak('Something went wrong').getResponse();
        };
  }

  toJSON(): InteractionModel {
    return this.model();
  }

  addCustomIntent(intent: Intent): this {
    if (this.intents[intent.props.intentName]) {
      throw new Error(`${intent.props.intentName} already defined`);
    }
    this.intents[intent.props.intentName] = intent;
    return this;
  }

  getString() {
    return this.localeValue;
  }

  model(): InteractionModel {
    const intents = Object.keys(this.intents).map((intentName: string) => this.intents[intentName]);
    const dialogIntents = intents.filter((intent) => intent.usesDialog());

    const dialog =
      dialogIntents.length > 0
        ? {
            intents: intents
              .filter((intent) => intent.usesDialog())
              .map((intent) => {
                const dialogIntents: DialogIntents = {
                  name: intent.props.intentName,
                  confirmationRequired: intent.props.confirmation !== undefined,
                  prompts: intent.props.confirmation
                    ? {
                        confirmation: hashValue(intent.props.confirmation.prompts)
                      }
                    : undefined,
                  delegationStrategy: intent.props.delegationStrategy,
                  slots: intent.slotsUsingDialog().map((slot) => {
                    const dialogSlot: DialogSlotItems = {
                      name: slot.name,
                      type: slot.type,
                      elicitationRequired: slot.required !== undefined,
                      confirmationRequired: slot.confirmation !== undefined,
                      prompts: {
                        elicitation: slot.required ? hashValue(slot.required.prompts) : undefined,
                        confirmation: slot.confirmation ? hashValue(slot.confirmation.prompts) : undefined
                      }
                    };
                    return dialogSlot;
                  })
                };
                return dialogIntents;
              })
          }
        : undefined;

    const prompts =
      dialogIntents.length > 0
        ? dialogIntents
            .flatMap((intent) => {
              const slotPrompts = intent.slotsUsingDialog().flatMap((slot) => {
                return [...(slot.confirmation ? [slot.confirmation.prompts] : []), ...(slot.required ? [slot.required.prompts] : [])];
              });
              const intentPrompts = intent.props.confirmation ? [intent.props.confirmation.prompts] : [];
              return [...slotPrompts, ...intentPrompts];
            })
            .map((prompt: PromptItems[]) => {
              return {
                id: hashValue(prompt),
                variations: prompt
              };
            })
        : undefined;

    return {
      interactionModel: {
        languageModel: {
          intents: intents
            .map((intent: Intent) => intent.model())
            .concat([
              {
                name: 'AMAZON.CancelIntent',
                samples: []
              },
              {
                name: 'AMAZON.StopIntent',
                samples: []
              },
              {
                name: 'AMAZON.HelpIntent',
                samples: []
              },
              {
                name: 'AMAZON.FallbackIntent',
                samples: []
              }
            ]),
          invocationName: this.props.invocationName
        },
        dialog,
        prompts
      }
    };
  }

  writeToPackageFile(baseSkillPackagePath?: string | undefined): void {
    const interactionFolder = (baseSkillPackagePath ? baseSkillPackagePath : './') + '/interactionModels/custom';
    if (!fs.existsSync(interactionFolder)) {
      fs.mkdirSync(interactionFolder, { recursive: true });
    }

    const model = this.model();
    // write interaction models
    fs.writeFileSync(`${interactionFolder}/${this.getString()}.json`, JSON.stringify(model, null, 2));
  }

  getErrorHandler(): LabeledErrorHandler {
    return {
      uniqueId: `ErrorHandler-${this.getString()}`,
      canHandle: (input: HandlerInput) => {
        const locale = getLocale(input.requestEnvelope);
        const isLocale = locale === this.getString();
        return isLocale;
      },
      handle: this.props.errorHandler!
    };
  }

  getRequestHandlers(): LabeledRequestHandler[] {
    const intentList = Object.keys(this.intents).map((intentName: string) => this.intents[intentName]!);

    return intentList
      .flatMap((intent: Intent) => intent.getRequestHandlers())
      .concat(
        {
          uniqueId: `SessionEnded-${this.getString()}`,
          canHandle: (input: HandlerInput) => {
            const locale = getLocale(input.requestEnvelope);
            const isLocale = locale === this.getString();
            if (!isLocale) {
              return false;
            }
            const requestType = getRequestType(input.requestEnvelope);
            const canHandle = requestType === 'SessionEndedRequest';

            return canHandle;
          },
          handle: wrapHandler(this.props.sessionEndedHandler!)
        },
        {
          uniqueId: `CancelHandler-${this.getString()}`,
          canHandle: (input: HandlerInput) => {
            const locale = getLocale(input.requestEnvelope);
            const isLocale = locale === this.getString();
            if (!isLocale) {
              return false;
            }
            const requestType = getRequestType(input.requestEnvelope);
            const canHandle =
              requestType === 'IntentRequest' &&
              (getIntentName(input.requestEnvelope) === 'AMAZON.StopIntent' ||
                getIntentName(input.requestEnvelope) === 'AMAZON.CancelIntent');

            return canHandle;
          },
          handle: wrapHandler(this.props.cancelHandler!)
        },
        {
          uniqueId: `FallbackHandler-${this.getString()}`,
          canHandle: (input: HandlerInput) => {
            const locale = getLocale(input.requestEnvelope);
            const isLocale = locale === this.getString();
            if (!isLocale) {
              return false;
            }
            const requestType = getRequestType(input.requestEnvelope);
            const canHandle = requestType === 'IntentRequest' && getIntentName(input.requestEnvelope) === 'AMAZON.FallbackIntent';
            return canHandle;
          },
          handle: wrapHandler(this.props.fallbackHandler!)
        },
        {
          uniqueId: `HelpHandler-${this.getString()}`,
          canHandle: (input: HandlerInput) => {
            const locale = getLocale(input.requestEnvelope);
            const isLocale = locale === this.getString();
            if (!isLocale) {
              return false;
            }
            const requestType = getRequestType(input.requestEnvelope);
            const canHandle = requestType === 'IntentRequest' && getIntentName(input.requestEnvelope) === 'AMAZON.HelpIntent';
            return canHandle;
          },
          handle: wrapHandler(this.props.helpHandler!)
        },
        {
          uniqueId: `LaunchHandler-${this.getString()}`,
          canHandle: (input: HandlerInput) => {
            const locale = getLocale(input.requestEnvelope);
            const isLocale = locale === this.getString();
            if (!isLocale) {
              return false;
            }
            const requestType = getRequestType(input.requestEnvelope);
            return requestType === 'LaunchRequest';
          },
          handle: wrapHandler(this.props.launchHandler!)
        }
      );
  }
}
