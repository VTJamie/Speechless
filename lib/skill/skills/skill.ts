import * as Alexa from 'ask-sdk';
import { DynamoDbPersistenceAdapter } from 'ask-sdk';
import { DynamoDB } from 'aws-sdk';
import * as fs from 'fs';

import { APLDocument } from '../../apl';
import { APLADocument } from '../../apla';
import { LocaleString, ModelProvider } from '../../types';
import { RegionAllAllowedTypesEndpointUri, SkillManifestEndpointAllAllowedTypesEndpointUri, SkillManifestModel } from '../interfaces';
import { Locale } from '../locale';
import { DynamoDBInfo, FilePackageWriter, LabeledRequestHandler } from '../models';
import { JSBRequestInterceptor, JSBResponseInterceptor } from '../types';

export type Locales = { [locale in LocaleString]?: Locale };
export type APLDocuments = { [documentName: string]: APLDocument };
export type APLADocuments = { [documentName: string]: APLADocument };
export interface CustomEndpointsProps {
  endpoint: SkillManifestEndpointAllAllowedTypesEndpointUri;
  regions?: {
    EU?: RegionAllAllowedTypesEndpointUri;
    FE?: RegionAllAllowedTypesEndpointUri;
    IN?: RegionAllAllowedTypesEndpointUri;
    NA?: RegionAllAllowedTypesEndpointUri;
  };
}

export interface SkillProps {
  name: string;
  custom: CustomEndpointsProps;
  customUserAgent?: string;
  dynamoDBInfo?: DynamoDBInfo;
  requestInterceptor?: JSBRequestInterceptor;
  responseInterceptor?: JSBResponseInterceptor;
}

export class Skill implements ModelProvider<SkillManifestModel>, FilePackageWriter {
  readonly props: SkillProps;
  readonly locales: Locales;
  readonly aplDocuments: APLDocuments;
  readonly aplaDocuments: APLADocuments;

  constructor(props: SkillProps) {
    this.locales = {};
    this.aplDocuments = {};
    this.aplaDocuments = {};
    this.props = props;
  }
  toJSON(): SkillManifestModel {
    return this.model();
  }

  addLocale(locale: Locale): this {
    if (this.locales[locale.localeValue]) {
      throw new Error(`${locale.localeValue} already added to skill`);
    }
    this.locales[locale.localeValue] = locale;
    return this;
  }

  addAPLDocument(aplDocument: APLDocument): this {
    if (this.aplDocuments[aplDocument.props.documentName]) {
      throw new Error(`APL Document already added ${aplDocument.props.documentName}`);
    }
    this.aplDocuments[aplDocument.props.documentName] = aplDocument;
    return this;
  }

  addAPLADocument(aplaDocument: APLADocument): this {
    if (this.aplDocuments[aplaDocument.props.documentName]) {
      throw new Error(`APLA Document already added ${aplaDocument.props.documentName}`);
    }
    this.aplaDocuments[aplaDocument.props.documentName] = aplaDocument;
    return this;
  }

  addAlexaHostedDynamoDBPersistenceAdapter(builder: Alexa.CustomSkillBuilder): void {
    if (this.props.dynamoDBInfo) {
      builder.withPersistenceAdapter(
        new DynamoDbPersistenceAdapter({
          tableName: process.env.DYNAMODB_PERSISTENCE_TABLE_NAME!,
          createTable: true,
          dynamoDBClient: new DynamoDB({ apiVersion: 'latest', region: process.env.DYNAMODB_PERSISTENCE_REGION! })
        })
      );
    }
  }
  addDynamoDBPersistenceAdapter(builder: Alexa.CustomSkillBuilder): void {
    if (this.props.dynamoDBInfo) {
      builder.withPersistenceAdapter(new DynamoDbPersistenceAdapter(this.props.dynamoDBInfo));
    }
  }

  addRequestInterceptor(builder: Alexa.CustomSkillBuilder): void {
    if (this.props.requestInterceptor) {
      builder.addRequestInterceptors({
        process: this.props.requestInterceptor
      });
    }
  }

  addResponseInterceptor(builder: Alexa.CustomSkillBuilder): void {
    if (this.props.responseInterceptor) {
      builder.addResponseInterceptors({
        process: this.props.responseInterceptor
      });
    }
  }

  getErrorHandlers(): Alexa.ErrorHandler[] {
    return this.getLocaleStrings()
      .map((ls) => this.locales[ls])
      .map((locale) => locale!.getErrorHandler());
  }

  builder(): Alexa.CustomSkillBuilder {
    const builder = Alexa.SkillBuilders.custom()
      .addRequestHandlers(...this.getRequestHandlers())
      .addErrorHandlers(...this.getErrorHandlers())

      .withCustomUserAgent(this.props.customUserAgent ? this.props.customUserAgent : 'JamiabboSkillBuilder');

    this.addRequestInterceptor(builder);
    this.addResponseInterceptor(builder);
    return builder;
  }

  handler(): Alexa.LambdaHandler {
    return this.builder().lambda();
  }

  writeSkillPackage(rootFolder?: string) {
    const baseSkillPackagePath = (rootFolder ? rootFolder : '.') + '/skill-package/';
    (Object.keys(this.locales) as LocaleString[])
      .map((locale: LocaleString) => this.locales[locale]!)
      .forEach((locale: Locale) => {
        locale.writeToPackageFile(baseSkillPackagePath);
      });
    Object.keys(this.aplDocuments)
      .map((documentName: string) => this.aplDocuments[documentName])
      .forEach((aplDocument: APLDocument) => aplDocument.writeToFile(baseSkillPackagePath));
    Object.keys(this.aplaDocuments)
      .map((documentName: string) => this.aplaDocuments[documentName])
      .forEach((aplaDocument: APLADocument) => aplaDocument.writeToFile(baseSkillPackagePath));

    this.writeToPackageFile(baseSkillPackagePath);
  }

  getRequestHandlers(): LabeledRequestHandler[] {
    const localeList = this.getLocaleStrings().map((locale: LocaleString) => this.locales[locale]!);
    const aplDocumentKeys = Object.keys(this.aplDocuments).map((documentName: string) => this.aplDocuments[documentName]);
    const requestHandlers = localeList
      .flatMap((locale: Locale) => locale.getRequestHandlers())
      .concat(aplDocumentKeys.flatMap((aplDocument: APLDocument) => aplDocument.getRequestHandlers()));
    return requestHandlers;
  }

  private getLocaleStrings(): LocaleString[] {
    return Object.keys(this.locales) as LocaleString[];
  }

  model(): SkillManifestModel {
    return {
      manifest: {
        manifestVersion: '1.0',

        apis: {
          custom: {
            dialogManagement: {
              dialogManagers: [
                {
                  type: 'AMAZON.Conversations'
                }
              ]
            },
            endpoint: this.props.custom?.endpoint,
            regions: this.props.custom?.regions,
            interfaces: [
              {
                type: 'ALEXA_PRESENTATION_APL',
                supportedViewports: [
                  {
                    mode: 'HUB',
                    shape: 'ROUND',
                    minWidth: 100,
                    maxWidth: 599,
                    minHeight: 100,
                    maxHeight: 599
                  },
                  {
                    mode: 'HUB',
                    shape: 'RECTANGLE',
                    minHeight: 600,
                    maxHeight: 959,
                    minWidth: 960,
                    maxWidth: 1279
                  },
                  {
                    mode: 'HUB',
                    shape: 'RECTANGLE',
                    minHeight: 600,
                    maxHeight: 1279,
                    minWidth: 1280,
                    maxWidth: 1920
                  },
                  {
                    mode: 'TV',
                    shape: 'RECTANGLE',
                    minHeight: 540,
                    maxHeight: 540,
                    minWidth: 960,
                    maxWidth: 960
                  }
                ]
              }
            ]
          }
        },
        publishingInformation: {
          locales: [
            {},
            ...this.getLocaleStrings().map((localeString: LocaleString) => {
              return {
                [localeString]: {
                  name: this.props.name
                }
              };
            })
          ].reduce((left, right) => {
            return {
              ...left,
              ...right
            };
          }),
          name: this.props.name
        }
      }
    };
  }

  writeToPackageFile(baseSkillPackagePath?: string | undefined): void {
    const packageFolder = baseSkillPackagePath ? baseSkillPackagePath : './';
    if (!fs.existsSync(packageFolder)) {
      fs.mkdirSync(packageFolder, { recursive: true });
    }

    const model = this.model();
    // write interaction models
    fs.writeFileSync(`${packageFolder}/skill.json`, JSON.stringify(model, null, 2));
  }
}
