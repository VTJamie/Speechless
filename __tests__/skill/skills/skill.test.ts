import { Locale, Skill } from '../../../lib/skill';
import { matchSnapshot } from '../../helper';
matchSnapshot(
  new Skill({
    name: 'TheBestSkill'
  }).addLocale(
    new Locale({
      locale: 'en-US',
      invocationName: 'test'
    })
  ),
  5
);
