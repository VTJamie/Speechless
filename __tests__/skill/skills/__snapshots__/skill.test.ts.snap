// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skill: Skill 1`] = `
{
  "manifest": {
    "apis": {
      "custom": {
        "dialogManagement": {
          "dialogManagers": [
            {
              "type": "AMAZON.Conversations",
            },
          ],
        },
        "interfaces": [
          {
            "supportedViewports": [
              {
                "maxHeight": 599,
                "maxWidth": 599,
                "minHeight": 100,
                "minWidth": 100,
                "mode": "HUB",
                "shape": "ROUND",
              },
              {
                "maxHeight": 959,
                "maxWidth": 1279,
                "minHeight": 600,
                "minWidth": 960,
                "mode": "HUB",
                "shape": "RECTANGLE",
              },
              {
                "maxHeight": 1279,
                "maxWidth": 1920,
                "minHeight": 600,
                "minWidth": 1280,
                "mode": "HUB",
                "shape": "RECTANGLE",
              },
              {
                "maxHeight": 540,
                "maxWidth": 960,
                "minHeight": 540,
                "minWidth": 960,
                "mode": "TV",
                "shape": "RECTANGLE",
              },
            ],
            "type": "ALEXA_PRESENTATION_APL",
          },
        ],
      },
    },
    "manifestVersion": "1.0",
    "publishingInformation": {
      "locales": {
        "en-US": {
          "name": "TheBestSkill",
        },
      },
      "name": "TheBestSkill",
    },
  },
}
`;
